django/core/files/storage.py

```diff
@@ -324,6 +339,25 @@ def created_time(self, name):
     def modified_time(self, name):
         return datetime.fromtimestamp(os.path.getmtime(self.path(name)))
 
+    def _do_safe_operation(self, operation, src_path, dst_path):
+        try:
+            operation(src_path, dst_path)
+        except OSError as e:
+            if e.errno == errno.EEXIST:
+                raise ValueError('Destination already exists')
```

Comment:
I don't like the idea of using `OSError` because the Storage interface is meant to be storage backend agnostic. You definitely wouldn't expect an `OSError` when copying or moving a file on S3.

I used `ValueError` as I see this as a generic way of flagging any invalid argument.

I also didn't like the idea of not specifying what the behavior should be when `dst_path` already exists. Having the API clients catching different types of exceptions depending on what storage backend they use would use would result in a leaky abstraction.

An alternative to using `ValueError` would be to create a custom exception type specific to django's storage API.

Thoughts?